contracts/sumtree-orderbook/src/sumtree/tree.rs

```diff
@@ -88,13 +90,40 @@ fn prefix_sum_walk(
         return Ok(current_sum);
     }
 
-    // --- Attempt walk left ---
-
     // We fetch both children here since we need to access both regardless of
     // whether we walk left or right.
     let left_child = node.get_left(storage)?;
     let right_child = node.get_right(storage)?;
 
+    // -- Resync Condition --
+
+    // To prevent requiring a resync there needs to be a condition that covers the case that
+    // when realizing the left node the new ETAS is enough to realize the right node (to some extent)
+    // To cover this we can determine how much of the left node has been realized, using this we can then determine
+    // if realizing what is unrealized from the left node will result in a new ETAS that is enough to realize the
+    // right node (to some extent)
+    if left_child.is_some() && right_child.is_some() {
+        let left_child = left_child.clone().unwrap();
+        let right_child = right_child.clone().unwrap();
+
+        // Calculate what the sum is before realizing the current node
+        let sum_at_node = current_sum.checked_sub(node.get_value())?;
+        // Calculate how much of the node has been realized in a previous sync
+        let diff_at_node = prev_sum.saturating_sub(sum_at_node);
+        // Calculate how much of the left node is unrealized
```

Comment:
```suggestion
        // `unrealized_from_left` is the amount in the left child that remained unrealized at the end of the previous
        // sync.
        //
        // If the left child is fully realized then the subtraction here will either be zero (if none of the right child is
        // realized) or negative (if some of the right child is realized), since `diff_at_node` = amount realized below left
        // and below right children. Saturating sub will ensure both of these cases snap to zero.
        // 
        // Thus, if this value is ever nonzero, it means that the left child had some unrealized cancels in it, and the
        // amount corresponds exactly to `unrealized_from_left`.
```